horust/example_services/sample_service.toml

```diff
@@ -48,3 +48,10 @@ wait = "10s"
 # If any of the services in the list has failed, shut down this service.
 die-if-failed = ["db.toml"]
 
+[resource-limit]
+# CPU quota. 0.5 means 50% time of one CPU core. -1 means no limit.
+cpu = 0.5
```

Comment:
thinking out loud, do you think it would be better to avoid floating point and just use decimal values? 
cpu = 50 would mean 50% of the time
cpu = 150 would mean 1 core and half.
empty would be no limit
and we could get rid of floating point math and conversions and it might also be intuitive from UX perspective. WDYT?

Reply:
Previously, I used `cpu-percent`. Later, I found that both Docker and Kubernetes use float. This might be the most widely used configuration format for CPU limit.

Meanwhile, `cpu-percent` is also supported by `docker run` on Windows.

I also asked GPT about this, float is more like the standard way. I'm also okay with changing it back to the `cpu-percent`.

- `docker run --cpus`: https://docs.docker.com/reference/cli/docker/container/run/
- `Kubernetes resource limit for CPU`: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu